draft-saml-ipsie-sl1-profile.md

```diff
@@ -0,0 +1,382 @@
+- shall require multi-factor authentication for all authentication events and encode this fact in the AuthnContextClassRef element;
+- shall maintain records of federation events for a minimum of 7 years or as required by applicable regulations;
+- shall implement a federation termination mechanism that allows for immediate revocation of federation relationships.
+
+#### 3.3.2. Requirements for Service Providers
+
+Service Providers:
+
+- shall support the SAML Web Browser SSO Profile as defined in [SAML2Profiles];
+- shall support verification of signatures using X.509 certificates;
+- shall support decryption of encrypted assertions using appropriate algorithms;
+- shall not expose open redirectors;
+- shall only use Identity Provider metadata retrieved from authoritative sources;
+- shall ensure that the metadata URL used is obtained from an authoritative source and using a secure channel, such that it cannot be modified by an attacker;
+- shall ensure that the entity ID used and the entity ID in the obtained metadata match;
+- shall implement federation termination procedures that can be executed promptly when required;
+- shall maintain session records that establish the link between the local session and the corresponding SAML assertion.
+
+Service Providers processing SAML assertions:
+
+- shall validate signatures on assertions and responses;
+- shall verify that the assertion was issued to them by checking the audience restriction;
+- shall check the validity period of the assertion;
+- shall implement replay detection to prevent assertion reuse;
+- shall validate the issuer of the assertion against trusted metadata;
+- shall verify holder-of-key bindings when present in the assertion;
+- shall enforce the session lifetime as specified in the assertion's SessionNotOnOrAfter attribute without exception;
+- shall terminate user sessions immediately when the session lifetime specified in the assertion expires;
+- shall not establish sessions with durations longer than specified in the SessionNotOnOrAfter attribute under any circumstances;
+- shall respect authentication context requirements and shall not grant access to protected resources if the authentication context does not meet requirements;
+- shall verify that multi-factor authentication was performed by checking the AuthnContextClassRef value;
+- shall implement appropriate security controls around any PII received in assertions;
+- shall implement appropriate subject identifier management that prevents correlation of subject identifiers across relying parties without user consent.
+
+For the Web Browser SSO Profile, Service Providers:
+
+- shall use signed authentication requests;
+- shall verify the InResponseTo attribute matches the ID of the request that was sent;
+- shall check the Issuer element in the SAML response to prevent mix-up attacks;
+- shall implement the SAML Enhanced Client or Proxy (ECP) Profile for non-browser clients;
```

Comment:
Do we really want to support this with IPSIE?  Feels like a great way to have a MFA downgrade attack especially as we want phishing resistant MFA